### PR TITLE
feat(utils): Fix big number decorator and cleanup

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
- Fix big number decorator and cleanup

### CODE REVIEW

**Review by riqwan — APPROVED** [review]
Much more cleaner. 🔥 +1 on adding mikroorm prefix

File: packages/utils/src/dal/mikro-orm/big-number-field.ts
Comment (fPolic) [inline packages/utils/src/dal/mikro-orm/big-number-field.ts]: question: aren't we setting `this[targetColumn]` on L:28 as a number so this check will never pass i.e. we will always call BigNumber constructor

Comment (adrien2p) [reply packages/utils/src/dal/mikro-orm/big-number-field.ts]: should be all good man 👍

File: packages/utils/src/dal/mikro-orm/big-number-field.ts
Comment (sradevski) [inline packages/utils/src/dal/mikro-orm/big-number-field.ts:5]: Since this is already under `mikro-orm` folder, I think naming it `BigNumberProperty` is sufficient, WDYT? It will make it a bit nicer to read, and the prefix feels unnecessary.

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Review by sradevski — COMMENTED** [review]
Let's make a final decision internally on how to handle money and big numbers before we merge this

File: packages/utils/src/dal/mikro-orm/big-number-field.ts
Comment (adrien2p) [reply packages/utils/src/dal/mikro-orm/big-number-field.ts:5]: I do agree, though, when you import it, you import from utils, not necessarily knowing where it is located in the package ☺️ so i preferred to be extra clear

**Comment by adrien2p** [comment]
> Let's make a final decision internally on how to handle money and big numbers before we merge this

This pr is meant to fix current issues, i think it could still be merged to resolve those issues in the different other pr's and we can come back to it when we settled? No strong opinion though

File: packages/utils/src/dal/mikro-orm/big-number-field.ts
Comment (riqwan) [reply packages/utils/src/dal/mikro-orm/big-number-field.ts:5]: Just for added context, the model/repository layer is designed to be completely replaceable. Which is why you'll see a lot of places have a MikroORM prefix.

**Comment by riqwan** [comment]
+1 this fixes an issue I have in my PR. We can still do a followup after we've aligned.

File: packages/payment/src/models/payment-session.ts
Comment (olivermrbl) [inline packages/payment/src/models/payment-session.ts]: question: Why have we added `number` here?

Comment (adrien2p) [reply packages/payment/src/models/payment-session.ts]: My mistake, completely involuntary

**Comment by olivermrbl** [comment]
I would also be OK with merging this now. It is blocking a few other PRs.

**Pull request merged**
